modules/engines/engine-javascript-graalium/src/main/java/org/eclipse/dirigible/engine/js/graalium/execution/eventloop/executor/AutoShutdownableLoopExecutor.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.dirigible.engine.js.graalium.execution.eventloop.executor;
+
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+public class AutoShutdownableLoopExecutor {
+
+    private final ScheduledExecutorService backgroundExecutor = Executors.newScheduledThreadPool(4);
+    private final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+    private final AtomicInteger tasksCounter = new AtomicInteger(0);
+    private boolean isInAwaitState = false;
```

Comment:
Is this needed, it seems to be used at one place only.

Reply:
It is used in order to not shutdown the executor service in `decrementTasksCount`. If `isInAwaitState` is not used, when we fire up the initial `eval`of a given JS code, the executor would shutdown once it evaluates the JS and not wait for setTimeouts and other kinds of timers

Reply:
Got it, thanks!